android/app/src/main/java/org/openbot/common/CameraFragment.java

```diff
@@ -62,8 +63,14 @@ private View addCamera(View view, LayoutInflater inflater, ViewGroup container)
     previewView = cameraView.findViewById(R.id.viewFinder);
     rootView.addView(view);
 
-    if (allPermissionsGranted()) setupCamera();
-    else requestPermissions(PERMISSIONS_REQUIRED, PERMISSIONS_REQUEST_CODE);
+    if (ContextCompat.checkSelfPermission(requireContext(), Constants.PERMISSION_CAMERA)
+        == PackageManager.PERMISSION_GRANTED) {
+      setupCamera();
+    } else if (shouldShowRequestPermissionRationale(Constants.PERMISSION_CAMERA)) {
+      PermissionUtils.showCameraPermissionsControllerToast(requireActivity());
```

Comment:
```
  public static void showCameraPermissionsControllerToast(Activity activity) {
    Toast.makeText(
            activity.getApplicationContext(),
            R.string.camera_permission_denied + " " + R.string.permission_reason_stream_video,
            Toast.LENGTH_LONG)
        .show();
  }
```
this is the definition of the function. just one toast.

Reply:
Yes but since it comes from camera fragment, we don't know if the permission was needed for the controller or something else. As a matter of fact it is probably used for previewing the camera feed isn't it?

Reply:
right, so you want to change the message? if so, can you share some content?